resources/sip11/ra/src/test/java/org/mobicents/slee/resource/sip11/test/SipResourceAdaptorTest.java

```diff
@@ -0,0 +1,144 @@
+		Whitebox.setInternalState(ra1, "providerWrapper", sleeSipProvider);
+
+		// initialize internal dependencies of RA#2
+		Whitebox.setInternalState(ra2, "tracer", tracer1);
+		Whitebox.setInternalState(ra2, "raContext", raContext2);
+		Whitebox.setInternalState(ra2, "providerWrapper", sleeSipProvider);
+
+		// prepare configuration for RA#1, RA#2
+		ConfigProperties configRa1 = new ConfigProperties();
+		ConfigProperties configRa2;
+
+		configRa1.addProperty(new Property("javax.sip.TRANSPORT", "java.lang.String", "TCP"));
+		configRa1.addProperty(new Property("javax.sip.IP_ADDRESS", "java.lang.String", STACK_ADDRESS));
+		configRa1.addProperty(new Property("org.mobicents.ha.javax.sip.BALANCERS", "java.lang.String", ""));
+		configRa1.addProperty(new Property("org.mobicents.ha.javax.sip.LoadBalancerHeartBeatingServiceClassName",
+				"java.lang.String", "org.mobicents.ha.javax.sip.LoadBalancerHeartBeatingServiceImpl"));
+		configRa1.addProperty(new Property("org.mobicents.ha.javax.sip.LoadBalancerElector", "java.lang.String",
+				"org.mobicents.ha.javax.sip.RoundRobinLoadBalancerElector"));
+		configRa1.addProperty(new Property("org.mobicents.ha.javax.sip.CACHE_CLASS_NAME", "java.lang.String",
+				"org.mobicents.ha.javax.sip.cache.NoCache"));
+
+		// clone common properties
+		configRa2 = (ConfigProperties) configRa1.clone();
+
+		// RA#1 custom configuration
+		configRa1.addProperty(new Property("javax.sip.PORT", "java.lang.Integer", SIP_RA1_PORT));
+		configRa1.addProperty(new Property("javax.sip.STACK_NAME", "java.lang.String", raContext1.getEntityName()));
+
+		// RA#2 custom configuration
+		configRa2.addProperty(new Property("javax.sip.PORT", "java.lang.Integer", SIP_RA2_PORT));
+		configRa2.addProperty(new Property("javax.sip.STACK_NAME", "java.lang.String", raContext2.getEntityName()));
+
+		// configure adaptors
+		ra1.raConfigure(configRa1);
+		Assert.assertEquals(SIP_RA1_PORT, Whitebox.getInternalState(ra1, "port"));
+		Assert.assertEquals(STACK_ADDRESS, Whitebox.getInternalState(ra1, "stackAddress"));
+
+		ra2.raConfigure(configRa2);
+		Assert.assertEquals(SIP_RA2_PORT, Whitebox.getInternalState(ra2, "port"));
+		Assert.assertEquals(STACK_ADDRESS, Whitebox.getInternalState(ra2, "stackAddress"));
+
+		ra1.raActive();
+		ra2.raActive();
+
+		// verify RA1 stack
+		ClusteredSipStack stack1 = Whitebox.getInternalState(ra1, "sipStack");
+		Assert.assertEquals(raContext1.getEntityName(),stack1.getStackName());
+
+		// verify that RA1 - stack is listening on a port used in configuration
+		Iterator stack1ListeningPoints = stack1.getListeningPoints();
+		while (stack1ListeningPoints.hasNext()) {
+			ListeningPoint listeningPoint = (ListeningPoint) stack1ListeningPoints.next();
+			Assert.assertEquals(SIP_RA1_PORT,listeningPoint.getPort());
+		}
+
+		// verify RA2 stack
+		ClusteredSipStack stack2 = Whitebox.getInternalState(ra2, "sipStack");
+		Assert.assertEquals(raContext2.getEntityName(),stack2.getStackName());
```

Comment:
Mock is not needed to get entity name, but sipStack field is private, therefor some API of powermock is used to help. More on that in the general remark on using Mock libraries.